### ISSUE TITLE
Request context not available in django >= 1.10

### ISSUE THREAD

**Issue by jxrossel** [TriOptima/tri.table#12]
Hi (again ;-) ),

I've had some trouble determining why `render_table_to_response` doesn't provide the request context in my template. I see in the code that `render_table` calls `template.render` with a `RequestContext` object given by `table_context`. 

According to this section of Django's doc, that looks fine: https://docs.djangoproject.com/en/1.11/ref/templates/api/#using-requestcontext

But in the [1.10 release notes](https://docs.djangoproject.com/en/1.11/releases/1.10/), I've found: 

> Django template objects returned by get_template() and select_template() no longer accept a Context in their render() method

... although no error message is raised !

The new solution is also provided [in the docs](https://docs.djangoproject.com/en/1.11/topics/templates/#template): `Template.render(context=None, request=None)`

**Comment by boxed** [comment]
The django docs seem pretty confused about this I must say! https://docs.djangoproject.com/en/1.11/topics/templates/#context explicitly says that's what Context objects is about :(

**Comment by jxrossel** [comment]
Hi,

I think I finally understood what's going on. According to this [overview](https://docs.djangoproject.com/en/1.11/ref/templates/api/#overview), the Django template engine is wrapped into a generic API that users should generally use.

The (low-level) Django template engine has an `Engine.get_template()` which returns a `Template` object that has a `.render(Context)` method with a `Context` (or `RequestContext`) object as input. All these things are actually wrapped in the generic API and the method signature changes (while keeping the same name...), yielding `.render( request, context_dict )`. So... the docs sometimes describe one and sometimes the other ! As far as I've understood, the behaviour therefore depends on the API chosen by the user (low-level - `Engine.get_template()` - or engine agnostic - `django.template.loader.get_template()` - (recommended)).

So basically, in order to use the generic API, one must:
- call `django.template.loader.get_template( 'template_name' ).render( context=pure_dict, request=request )`
- link with the custom context processors in the settings file (cannot be given explicitly here)

I've made some tests and it works. If `.render()` gets a `request` in its input, it will internally call `RequestContext`. The given context must be a pure python dictionary.

**Comment by boxed** [comment]
Yea, that's what I'm reading too. Extremely confusing! The problem is that your fix only works for django 1.10+, while we support older versions... so I need to write code that handles all the versions :( working on this right now...

**Comment by jxrossel** [comment]
Yep, that took me a while to un-confuse it. Thanks !

**Comment by boxed** [comment]
I have a fix for this.. just waiting for code review internally here and I'll push out a new release for you.

**Comment by jxrossel** [comment]
Good news! Thanks a lot!

**Comment by boxed** [comment]
tri.table 4.3.0 is on pypi now, let me know if it works for you, then I'll close this issue.

**Comment by jxrossel** [comment]
Well, you nailed it. It works like a charm. Thanks again!

**Comment by boxed** [comment]
Awesome. Thanks for trying it out and finding these bugs :P
